lib/validator/validator.go

```diff
@@ -36,22 +36,28 @@ import (
 var validate *validator.Validate
 
 var (
-	labelRegex          = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `$`)
-	labelValueRegex     = regexp.MustCompile("^[a-zA-Z0-9]?([a-zA-Z0-9_.-]{0,61}[a-zA-Z0-9])?$")
-	nameRegex           = regexp.MustCompile("^[a-zA-Z0-9_.-]{1,128}$")
-	namespacedNameRegex = regexp.MustCompile(`^[a-zA-Z0-9_./-]{1,128}$`)
-	interfaceRegex      = regexp.MustCompile("^[a-zA-Z0-9_-]{1,15}$")
-	actionRegex         = regexp.MustCompile("^(allow|deny|log|pass)$")
-	backendActionRegex  = regexp.MustCompile("^(allow|deny|log|next-tier|)$")
-	protocolRegex       = regexp.MustCompile("^(tcp|udp|icmp|icmpv6|sctp|udplite)$")
-	ipipModeRegex       = regexp.MustCompile("^(always|cross-subnet|)$")
-	reasonString        = "Reason: "
-	poolSmallIPv4       = "IP pool size is too small (min /26) for use with Calico IPAM"
-	poolSmallIPv6       = "IP pool size is too small (min /122) for use with Calico IPAM"
-	poolUnstictCIDR     = "IP pool CIDR is not strictly masked"
-	overlapsV4LinkLocal = "IP pool range overlaps with IPv4 Link Local range 169.254.0.0/16"
-	overlapsV6LinkLocal = "IP pool range overlaps with IPv6 Link Local range fe80::/10"
-	protocolPortsMsg    = "rules that specify ports must set protocol to TCP or UDP"
+	labelRegex            = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `$`)
+	labelValueRegex       = regexp.MustCompile("^[a-zA-Z0-9]?([a-zA-Z0-9_.-]{0,61}[a-zA-Z0-9])?$")
+	nameRegex             = regexp.MustCompile("^[a-zA-Z0-9_.-]{1,128}$")
+	labelsToApplyValRegex = regexp.MustCompile("^[a-zA-Z0-9_.-]{0,128}$")
+	namespacedNameRegex   = regexp.MustCompile(`^[a-zA-Z0-9_./-]{1,128}$`)
+	interfaceRegex        = regexp.MustCompile("^[a-zA-Z0-9_-]{1,15}$")
+	actionRegex           = regexp.MustCompile("^(allow|deny|log|pass)$")
+	backendActionRegex    = regexp.MustCompile("^(allow|deny|log|next-tier|)$")
+	protocolRegex         = regexp.MustCompile("^(tcp|udp|icmp|icmpv6|sctp|udplite)$")
+	ipipModeRegex         = regexp.MustCompile("^(Always|CrossSubnet|Never)$")
+	bgpLogLevel           = regexp.MustCompile("^(Info|Warning|Debug|Error|Fatal|Trace)$")
+	felixLogLevel         = regexp.MustCompile("^(Info|Warning|Debug|Error|Critical)$")
+	datastoreType         = regexp.MustCompile("^(etcdv3|kubernetes)$")
+	dropacceptreturnRegex = regexp.MustCompile("^(Drop|Accept|Return)$")
```

Comment:
On https://docs.projectcalico.org/master/reference/calicoctl/resources/felixconfig, drop/accept/return is denoted case-insensitive. The regex isn't case insensitive, what's the expected behavior?

Reply:
Yup, we need to change the doc so the yaml config follows the standard casing that's in this regex, once this PR is approved we'll need to update the doc page. I'll create a task for that, thanks @bcreane

Reply:
There are also some doc changes following from Rob's tweak PR: https://github.com/projectcalico/libcalico-go/pull/664. I can take the combined stories unless someone else is hankering for that.

Reply:
Good idea @bcreane, you can take it. Thanks!